osmoutils/accum/accum_test.go

```diff
@@ -318,3 +321,439 @@ func (suite *AccumTestSuite) TestClaimRewards() {
+		accumInit	sdk.DecCoins
+		accumDelta  sdk.DecCoins
+
+		// Address does not exist
+		addrDNE bool
+		expPass bool
+	}
+
+	tests := map[string]testcase{
+		"no new rewards": {
+			startingNumShares: initialValueOne,
+			startingUnclaimedRewards: sdk.NewDecCoins(),
+			removedShares: sdk.OneDec(),
+			accumInit: sdk.NewDecCoins(),
+			// unchanged accum value, so no unclaimed rewards
+			accumDelta: sdk.NewDecCoins(),
+			expPass: true,
+		},
+		"new rewards in one denom": {
+			startingNumShares: initialValueOne,
+			startingUnclaimedRewards: sdk.NewDecCoins(),
+			removedShares: sdk.OneDec(),
+			accumInit: sdk.NewDecCoins(),
+			// unclaimed rewards since last update
+			accumDelta: sdk.NewDecCoins(initialCoinDenomOne),
+			expPass: true,
+		},
+		"new rewards in two denoms": {
+			startingNumShares: initialValueOne,
+			startingUnclaimedRewards: sdk.NewDecCoins(),
+			removedShares: sdk.OneDec(),
+			accumInit: sdk.NewDecCoins(),
+			accumDelta: sdk.NewDecCoins(initialCoinDenomOne, initialCoinDenomTwo),
+			expPass: true,
+		},
+		"both existing and new rewards": {
+			startingNumShares: initialValueOne,
+			startingUnclaimedRewards: sdk.NewDecCoins(sdk.NewDecCoin(denomOne, sdk.NewInt(11)), sdk.NewDecCoin(denomTwo, sdk.NewInt(11))),
+			removedShares: sdk.OneDec(),
+			accumInit: sdk.NewDecCoins(),
+			accumDelta: sdk.NewDecCoins(initialCoinDenomOne, initialCoinDenomTwo),
+			expPass: true,
+		},
+		"both existing (one denom) and new rewards (two denoms, one overlapping)": {
+			startingNumShares: initialValueOne,
+			startingUnclaimedRewards: sdk.NewDecCoins(initialCoinDenomOne),
+			removedShares: sdk.OneDec(),
+			accumInit: sdk.NewDecCoins(),
+			accumDelta: sdk.NewDecCoins(initialCoinDenomOne, initialCoinDenomTwo),
+			expPass: true,
+		},
+		"both existing (one denom) and new rewards (two new denoms)": {
+			startingNumShares: initialValueOne,
+			startingUnclaimedRewards: sdk.NewDecCoins(initialCoinDenomOne),
+			removedShares: sdk.OneDec(),
+			accumInit: sdk.NewDecCoins(),
+			accumDelta: sdk.NewDecCoins(initialCoinDenomTwo, sdk.NewDecCoin("baz", sdk.NewInt(10))),
+			expPass: true,
+		},
+		"nonzero accumulator starting value, delta with same denoms": {
+			startingNumShares: initialValueOne,
+			startingUnclaimedRewards: sdk.NewDecCoins(initialCoinDenomOne),
+			removedShares: sdk.OneDec(),
+			accumInit: sdk.NewDecCoins(initialCoinDenomOne, initialCoinDenomTwo),
+			accumDelta: sdk.NewDecCoins(initialCoinDenomOne, initialCoinDenomTwo),
+			expPass: true,
+		},
+		"nonzero accumulator starting value, delta with new denoms": {
+			startingNumShares: initialValueOne,
+			startingUnclaimedRewards: sdk.NewDecCoins(initialCoinDenomOne),
+			removedShares: sdk.OneDec(),
+			accumInit: sdk.NewDecCoins(initialCoinDenomOne, initialCoinDenomTwo),
+			accumDelta: sdk.NewDecCoins(initialCoinDenomTwo, sdk.NewDecCoin("baz", sdk.NewInt(10))),
+			expPass: true,
+		},
+		"remove decimal shares with new rewards in two denoms": {
```

Comment:
Do we have a test case that covers the TODO for this method? That is, where we remove all shares from a position, making it obsolete

Reply:
Will add this functionality to `ClaimRewards` in a separate PR